test/gst-vaapi/util.py

```diff
@@ -75,6 +75,23 @@ def have_gst_element(element):
   result = try_call("gst-inspect-1.0 {}".format(element))
   return result, element
 
+@memoize
+def iHD_input_format(format):
```

Comment:
Please make this more general.  That is, do driver check inside this function (using media plugin's method) and only maintain formats that must be mapped for driver.  Also, we should map the normalized (e.g. user config) name to the middleware specific name.

For example:

```
@memoize
def mapformat_hwup(format):
  from ...lib import get_media
  fmt = {
    "iHD" : {
      "I420", "NV12",
    },
  }.get(get_media()._get_driver_name(), {}).get(format, format)

  return mapformatu(format)
```

Reply:
has been changed